Docs: External CA/PKI: clarify intermediate CA cross-signing options (#10392)
Icinga/icinga2: doc/06-distributed-monitoring.md

* Each side has to provide its intermediate CAs along with the leaf certificate
  in `/var/lib/icinga2/certs/NODENAME.crt`, ordered from leaf to root.
* Intermediate CAs may not be used directly as root CAs.

##### Using an intermediate CA as Icinga root CA

For Icinga to trust only its own intermediate CA,
do either of the following:

> ###### Icinga itself issues leaf certificates

Reviewer: ```suggestion
###### Leaf certificates are issued by Icinga
```

To match the passive tone of the other headline below.

Author: Feel free to press "Commit suggestion" and trigger GHA once everything else is ✅.

Author: Done.

* Each side has to provide its intermediate CAs along with the leaf certificate
  in `/var/lib/icinga2/certs/NODENAME.crt`, ordered from leaf to root.
* Intermediate CAs may not be used directly as root CAs.

##### Using external intermediate CA as Icinga root CA

For Icinga to trust only its own CA, if the latter shall be an intermediate one,

> do either of the following, depending on how strict your company policy is:

Author: If you have Root -> Int<sub>1</sub> -> Leaf, in a sane setup, you'd probably just use Root as Icinga Root CA. But then Icinga trusts Int<sub>n</sub>. Some people want **Icinga to trust only Int<sub>1</sub>**, e.g https://github.com/Icinga/icinga2/issues/7719#issuecomment-1496177500. But, as already in our docs, OpenSSL doesn't support an intermediate CA as root one, so you have to cross-sign.

But you can do latter in two directions, this is what I'm clarifying here.

> ###### Lax policy, Icinga itself issues leave certificates

Reviewer: Why is this policy lax and the other strict? This has some implications, potentially leading people to irrational decisions. I would suggest dropping the rating.

```suggestion
###### Leaf certificates issued by Icinga
```

> 1. Setup Icinga as usual, with its own CA issueing leave certificates

Reviewer: ```suggestion
1. Setup Icinga as usual, with its own CA issuing leaf certificates.
```

> 2. Cross-sign that CA with the desired parent CA, so that you get an intermediate CA
>    with the same subject and public key as the Icinga-own root CA

Reviewer: ```suggestion
2. Cross-sign this CA with the desired parent CA to create an intermediate CA.
```

> 3. Add that new intermediate CA to your trusted root CAs whereever possible and needed
>    to have an uninterrupted chain from your company CA to Icinga leave certificates

Reviewer: ```suggestion
3. Add that new intermediate CA to your trusted root CAs where needed
   to have an uninterrupted chain from your main CA to Icinga leaf certificates.
```

> ###### Strict policy, Icinga doesn't issue leave certificates

Reviewer: ```suggestion
###### Leaf certificates issued externally
```

1. Create your intermediate CA for Icinga
2. Cross-sign it with itself, so that you get two equal certificates,
   except that one is self-signed
3. Use the latter as Icinga root CA and deploy leave certificates manually,

>    each with the intermediate CA as described in the parent section

Author: @mkayontour had some questions on this section. While answering, I thought of an option to suggest, but then realized it's not documented yet.😅 So before I explained it in 🇩🇪, I thought let's just write it down in 🇬🇧 here:

>    each with the intermediate CA as described in the parent section

Reviewer: Please end each sentence with a period and change _leave_ to _leaf_, as I have suggested above.

Furthermore, this route needs a bit more details, imo. First, how should the intermediate CA be created? Then, unless I am mistaken, how can a CA cross-sign itself? Isn't this self-signing? When referring to "the latter" in your third statement, do you mean the signed one? Please name it.

Author: I'll have a look.

> First, how should the intermediate CA be created?

* We already decided in #9825 not to train people to do this

Reviewer: IMHO, this still has very high ["draw the rest of the owl" energy](https://knowyourmeme.com/photos/572078-how-to-draw-an-owl).
